proxy/hdrs/HTTP.cc

```diff
@@ -1133,7 +1133,12 @@ http_parser_parse_req(HTTPParser *parser, HdrHeap *heap, HTTPHdrImpl *hh, const
     ink_assert(hh->u.req.m_url_impl != nullptr);
 
     url = hh->u.req.m_url_impl;
-    err = ::url_parse(heap, url, &url_start, url_end, must_copy_strings, strict_uri_parsing);
+    if (method_wks_idx == HTTP_WKSIDX_OPTIONS && url_end - url_start == 1 && url_start[0] == '*') {
```

Comment:
Why do you use the magic `ASTERISK` value rather than just check for `*`?

Reply:
In URLImpl, having "*" in the path means the URL is "/*". The magic value is needed to distinguish the two.